### PR TITLE
media-gfx/sigal: update to new major release 2.0

### PR DESCRIPTION
**Pull request by tg--**
This brings in some new USE-flags (mainly for plugins), as well as
compatibility to Python 3.7.
Python < 3.6 is no longer supported by upstream.
Other than that, Sigal 2.0 is largely a drop-in-replacement for the 1.X
series.

Closes: https://bugs.gentoo.org/718334

Signed-off-by: Thomas Gstädtner <thomas@gstaedtner.net>

### CODE REVIEW

File: media-gfx/sigal/sigal-2.0.ebuild [hunk 1]
```diff
@@ -0,0 +1,42 @@
+# Copyright 1999-2020 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+PYTHON_COMPAT=( python3_{6,7} )
+
+inherit distutils-r1
+
+DESCRIPTION="Simple static web gallery generator"
+HOMEPAGE="http://sigal.saimon.org/"
```
Comment (charIes17) [inline media-gfx/sigal/sigal-2.0.ebuild:10]: HOMEPAGE could use https

Comment (tg--) [reply media-gfx/sigal/sigal-2.0.ebuild]: Agree, fixed.

File: media-gfx/sigal/sigal-2.0.ebuild [hunk 2]
```diff
@@ -0,0 +1,42 @@
+# Copyright 1999-2020 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+PYTHON_COMPAT=( python3_{6,7} )
```
Comment (juippis) [inline media-gfx/sigal/sigal-2.0.ebuild:5]: One empty line below EAPI please.

File: media-gfx/sigal/sigal-2.0.ebuild [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+HOMEPAGE="https://sigal.saimon.org/"
+SRC_URI="https://github.com/saimn/${PN}/archive/${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="MIT"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="brotli crypt rss s3 test zopfli"
+RESTRICT="!test? ( test )"
+
+CDEPEND="dev-python/blinker[${PYTHON_USEDEP}]
```
Comment (juippis) [inline media-gfx/sigal/sigal-2.0.ebuild:19]: Use `COMMON_DEPEND` instead.

Although this ebuild has no reason to use `CDEPEND` (or `COMMON_DEPEND`).

File: media-gfx/sigal/sigal-2.0.ebuild [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+inherit distutils-r1
+
+DESCRIPTION="Simple static web gallery generator"
+HOMEPAGE="https://sigal.saimon.org/"
+SRC_URI="https://github.com/saimn/${PN}/archive/${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="MIT"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="brotli crypt rss s3 test zopfli"
```
Comment (juippis) [inline media-gfx/sigal/sigal-2.0.ebuild:16]: Your USE flags have no function in the ebuild, and are banned by QA.
https://projects.gentoo.org/qa/policy-guide/dependencies.html#pg0001

File: media-gfx/sigal/sigal-2.0.ebuild [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+	rss? ( dev-python/feedgenerator[${PYTHON_USEDEP}] )
+	s3? ( dev-python/boto[${PYTHON_USEDEP}] )
+	test? (
+		dev-python/boto[${PYTHON_USEDEP}]
+		dev-python/pytest[${PYTHON_USEDEP}]
+	)
+	zopfli? ( app-arch/zopfli )"
+RDEPEND="${CDEPEND}"
+
+DOCS="README.rst"
```
Comment (juippis) [inline media-gfx/sigal/sigal-2.0.ebuild:38]: This should be an array. I also believe the default `einstalldocs` calls finds this file.

File: media-gfx/sigal/sigal-2.0.ebuild [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+
+python_test() {
+	esetup.py test
+}
```
Comment (juippis) [inline media-gfx/sigal/sigal-2.0.ebuild:42]: Use the function from *distutils-r1.eclass* instead.

**Comment by thesamesam** [comment]
ping

**Comment by mgorny** [comment]
I'm going to remove the package now. However, if you're still interested feel free to readd it via PR.

**Pull request closed without merging**